### ISSUE TITLE
Exception while saving context

### ISSUE THREAD

**Issue by vilinskiy-playdayteam** [RestKit/RestKit#2455]
Following exceptions are thrown while saving context
The first:
An observer of NSManagedObjectContextDidSaveNotification illegally threw an exception. 
```objc
 Objects saved = { 
    inserted = "{(\n)}"; 
    managedObjectContext = "<_PFWeakReference: 0x17042c0a0>"; 
    updated = "{(\n    <PDRecentItem: 0x1742cc4e0> (entity: PDRecentItem; id: 0xd000000001340014 <x-coredata://CF2A3944-65F8-4B04-A7EB-CB8CDA2607A2/PDRecentItem/p77> ; data: {\n    avatar = nil;\n    badgeIds = \"(\\n    5016517\\n)\";\n    chatContainerId = 2756;\n    chatContainerType = lobby;\n    chatId = 2756;\n    circleId = 650;\n    circleTitle = \"\\U041e\\U0438\\U043e\\U0438\\U0437\\U043e\\U0449\\U0438\\U0438\\U0449\\U0433\";\n    creationDate = \"2016-09-14 11:57:53 +0000\";\n    dialogOpponentName = nil;\n    dialogOpponentProfileId = nil;\n    hasAdminIcon = 1;\n    isPublic = 0;\n    lastMessageAuthorName = \"\\U041c\\U0430\\U0440\\U0438\\U043d\\U0430 \\U0438 \\U043a\\U043e\\U0442\\U0438\\U043a\\U0438\";\n    lastMessageBanned = 0;\n    lastMessageDeleted = 0;\n    lastMessageIsAuthorMe = 1;\n    lastMessageText = nil;\n    lastMessageType = \"join_room\";\n    latestListMeaningMessageId = 67106;\n    postAuthorName = nil;\n    postIsPublic = nil;\n    postMessageText = nil;\n    serverHash = nil;\n}),\n    <PDRecentItem: 0x1702d9830> (entity: PDRecentItem; id: 0xd000000001080014 <x-coredata://CF2A3944-65F8-4B04-A7EB-CB8CDA2607A2/PDRecentItem/p66> ; data: {\n    avatar = nil;\n    badgeIds = \"(\\n)\";\n    chatContainerId = 2762;\n    chatContainerType = lobby;\n    chatId = 2762;\n    circleId = 656;\n    circleTitle = \"\\U0418\\U0433\\U043e\\U0438\\U0448\\U0438\\U0448\\U0438\\U0449\\U044c\";\n    creationDate = \"2016-09-14 12:12:09 +0000\";\n    dialogOpponentName = nil;\n    dialogOpponentProfileId = nil;\n    hasAdminIcon = 1;\n    isPublic = 0;\n    lastMessageAuthorName = \"\\U041c\\U0430\\U0440\\U0438\\U043d\\U0430 \\U0438 \\U043a\\U043e\\U0442\\U0438\\U043a\\U0438\";\n    lastMessageBanned = 0;\n    lastMessageDeleted = 0;\n    lastMessageIsAuthorMe = 1;\n    lastMessageText = nil;\n    lastMessageType = \"join_room\";\n    latestListMeaningMessageId = 67117;\n    postAuthorName = nil;\n    postIsPublic = nil;\n    postMessageText = nil;\n    serverHash = nil;\n}),\n    <PDRecentItem: 0x1742cc710> (entity: PDRecentItem; id: 0xd000000001140014 <x-coredata://CF2A3944-65F8-4B04-A7EB-CB8CDA2607A2/PDRecentItem/p69> ; data: {\n    avatar = nil;\n    badgeIds = \"(\\n)\";\n    chatContainerId = 2758;\n    chatContainerType = lobby;\n    chatId = 2758;\n    circleId = 652;\n    circleTitle = \"\\U0428\\U0435\\U0438\\U043e\\U0448\\U0438\\U0440\\U0438\\U0448\";\n    creationDate = \"2016-09-14 12:06:10 +0000\";\n    dialogOpponentName = nil;\n    dialogOpponentProfileId = nil;\n    hasAdminIcon = 1;\n    isPublic = 0;\n    lastMessageAuthorName = \"\\U041c\\U0430\\U0440\\U0438\\U043d\\U0430 \\U0438 \\U043a\\U043e\\U0442\\U0438\\U043a\\U0438\";\n    lastMessageBanned = 0;\n    lastMessageDeleted = 0;\n    lastMessageIsAuthorMe = 1;\n    lastMessageText = nil;\n    lastMessageType = \"join_room\";\n    latestListMeaningMessageId = 67108;\n    postAuthorName = nil;\n    postIsPublic = nil;\n    postMessageText = nil;\n    serverHash = nil;\n})\n)}"; 
} and exception = [<_PFWeakReference 0x17042c0a0> valueForUndefinedKey:]: this class is not key value coding-compliant for the key @count. with userInfo = { 
    NSTargetObjectUserInfoKey = "<_PFWeakReference: 0x17042c0a0>"; 
    NSUnknownUserInfoKey = "@count"; 
}
```

The second one
```objc
An observer of NSManagedObjectContextDidSaveNotification illegally threw an exception.  Objects saved = { 
    inserted = "{(\n)}"; 
    managedObjectContext = "<_PFWeakReference: 0x174825ce0>"; 
    updated = "{(\n    <PDCircle: 0x1700d7920> (entity: PDCircle; id: 0xd0000000017c0006 <x-coredata://CF2A3944-65F8-4B04-A7EB-CB8CDA2607A2/PDCircle/p95> ; data: {\n    address = \"\\U0443\\U043b\\U0438\\U0446\\U0430 \\U0413\\U043e\\U0441\\U043f\\U0438\\U0442\\U0430\\U043b\\U044c\\U043d\\U044b\\U0439 \\U0412\\U0430\\U043b, 5 \\U0441\\U0442\\U0440\\U043e\\U0435\\U043d\\U0438\\U0435 5, \\U041c\\U043e\\U0441\\U043a\\U0432\\U0430, \\U0420\\U043e\\U0441\\U0441\\U0438\\U044f\";\n    adminProfileId = 2743;\n    avatar = nil;\n    badges =     (\n    );\n    chatId = 2757;\n    circleId = 651;\n    circleListItems = \"<relationship fault: 0x170635d00 'circleListItems'>\";\n    circleRecommendation = \"<relationship fault: 0x170635d40 'circleRecommendation'>\";\n    hidden = 1;\n    latitude = \"55.774445741879\";\n    lobby = \"0xd00000000054000c <x-coredata://CF2A3944-65F8-4B04-A7EB-CB8CDA2607A2/PDLobby/p21>\";\n    lobbyId = 2757;\n    longitude = \"37.709360606968\";\n    mapItem = nil;\n    outcastCount = 0;\n    postTimelineItems = \"<relationship fault: 0x170635e00 'postTimelineItems'>\";\n    profileCount = 1;\n    profiles =     (\n        \"0xd000000000c80010 <x-coredata://CF2A3944-65F8-4B04-A7EB-CB8CDA2607A2/PDProfile/p50>\"\n    );\n    serverHash = a1a1e7785e7f0b71ad1f24929644ec4e;\n    title = \"\\U0418\\U043e\\U0442\\U043e\\U0442\\U0433\\U043e\\U0433\\U043e\\U0448\";\n})\n)}"; 
} and exception = [<_PFWeakReference 0x174825ce0> valueForUndefinedKey:]: this class is not key value coding-compliant for the key objectID. with userInfo = { 
    NSTargetObjectUserInfoKey = "<_PFWeakReference: 0x174825ce0>"; 
    NSUnknownUserInfoKey = objectID;
```

The key to the problem in new NSManagedObjectContextDidSaveNotification.userInfo key
Now it contains managedObjectContext key

**Comment by segiddins** [comment]
This should  now be fixed on master

**Comment by vilinskiy-playdayteam** [comment]
Ok, I'll check now, thanks)

**Comment by vilinskiy-playdayteam** [comment]
Fixed in #2449
